tools/SampleAnalyzer/Process/Plot/HistoLogX.h

```diff
@@ -1,178 +1,122 @@
-
-    // Negative weight
-    else
-    {
-      nentries_.second++;
-      weight=std::abs(weight);
-      sum_w_.second  += weight;
-      sum_ww_.second += weight*weight;
-      sum_xw_.second += value*weight;
-      if (value < xmin_) underflow_.second+=weight;
-      else if (value >= xmax_) overflow_.second+=weight;
-      else
-      {
-        histo_[std::floor((std::log10(value)-log_xmin_)/step_)].second+=weight;
-      }
-    }
-  }
-
-
-  /// Write the plot in a Text file
-  virtual void Write_TextFormat(std::ostream* output);
-
-  // Write the plot in a ROOT file
-  //  virtual void Write_RootFormat(std::pair<TH1F*,TH1F*>& histos);
-
-};
+        // -------------------------------------------------------------
+        //                        data members
+        // -------------------------------------------------------------
+    protected:
+        // Histogram boundaries in Log scale
+        MAfloat64 log_xmin_;
+        MAfloat64 log_xmax_;
+
+        // -------------------------------------------------------------
+        //                       method members
+        // -------------------------------------------------------------
+    public:
+        /// Constructor withtout argument
+        HistoLogX() { initialised_ = false; }
+
+        /// Constructor with argument
+        HistoLogX(const std::string &name, MAuint32 nbins,
+                  MAfloat64 xmin, MAfloat64 xmax) : Histo(name)
+        {
+            // Setting the description: nbins
+            try
+            {
+                if (nbins == 0)
+                    throw EXCEPTION_WARNING("nbins cannot be equal to 0. Set 100.", "", 0);
+                nbins_ = nbins;
+            }
+            catch (const std::exception &e)
+            {
+                MANAGE_EXCEPTION(e);
+                nbins_ = 100;
+            }
+
+            // Setting the description: min
+            try
+            {
+                if (xmin <= 0)
+                    throw EXCEPTION_WARNING("xmin cannot be less than or equal to zero. Setting xmin to 0.1", "", 0);
+                xmin_ = xmin;
+            }
+            catch (const std::exception &e)
+            {
+                MANAGE_EXCEPTION(e);
+                xmin_ = .1;
+            }
+
+            // Setting the description: max
+            try
+            {
+                if (xmin >= xmax)
+                    throw EXCEPTION_WARNING("xmin cannot be equal to or greater than xmax. Setting xmin to 0.1 and xmax to 100.", "", 0);
+                xmax_ = xmax;
+            }
+            catch (const std::exception &e)
+            {
+                MANAGE_EXCEPTION(e);
+                xmin_ = .1;
+                xmax_ = 100.;
+            }
+
+            log_xmin_ = std::log10(xmin_);
+            log_xmax_ = std::log10(xmax_);
+            step_ = (log_xmax_ - log_xmin_) / static_cast<MAfloat64>(nbins_);
+
+            // Reseting the histogram array
+            histo_.resize(nbins_);
```

Comment:
Do we need to resize? I don't think so.

Reply:
I think it's for memory efficiency, suggested use of vectors.

Reply:
This was comment out in Histo.h. Why?

Reply:
I might have forgotten it, will check it out